### PR TITLE
Continuous Integration... 43 commits later

### PR DESCRIPTION
**Pull request by TheGlitch76**
Featuring:
- Auto build and checkstyle (as seperate tasks) for commits to any branch and pull requests, along with commits on forks.
- Auto version selection via the following criteria: the tag version (minus the v, so `v1.0.0` -> `1.0.0`) OR if that is not present the commit hash (i.e. `eb741d1`) OR 'LOCAL' as a fallback for not being on Github
- Auto publication on creation of a tag to Bintray of the latest build. Requires the secrets `bintray_user` and `bintray_key` to be set in Settings -> Secrets to function properly.
Issues:
- Tagged commits will have the checkstyle and build tests run twice (once for being a tag and once for being on the branch). There is no easy way to fix this because described behavior on Github of how tags can be ignored does not actually work.

### CODE REVIEW

**Comment by TheGlitch76** [comment]
Please squash commit this

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ data/
 .idea/
 build
 .gradle
-run/
\ No newline at end of file
+run/
+# TODO REMOVE ME
```
Comment (coderbot16) [inline .gitignore]: Ummm.....

File: build.gradle [hunk 2]
```diff
@@ -1,28 +1,35 @@
-apply plugin: 'checkstyle'
-apply plugin: 'application'
-apply plugin: 'eclipse'
-apply plugin: 'idea'
-apply plugin: 'maven-publish'
-
+plugins {
```
Comment (coderbot16) [inline build.gradle:1]: The indentation in this file seems very weird. Try a `Reformat Code`?

File: build.gradle [hunk 3]
```diff
@@ -69,3 +76,91 @@ if(!runDir.exists() && !runDir.mkdirs()) {
...
+
+			// add all the jars that should be included when publishing to maven
+			afterEvaluate {
+				artifact jar
+				artifact sourcesJar
+			}
+		}
+	}
+
+//	if (!isNullOrEmpty(System.getenv('GITHUB_OAUTH_TOKEN'))) {
```
Comment (coderbot16) [inline build.gradle]: Maybe remove this for now?

File: build.gradle [hunk 4]
```diff
@@ -69,3 +76,91 @@ if(!runDir.exists() && !runDir.mkdirs()) {
...
+		websiteUrl = "https://github.com/${project.repoOwner}/${project.repoName}"
+		issueTrackerUrl = "https://github.com/${project.repoOwner}/${project.repoName}/issues"
+		vcsUrl = "https://github.com/${project.repoOwner}/${project.repoName}.git"
+		publicDownloadNumbers = true
+
+		githubRepo = "${project.repoOwner}/${project.repoName}"
+		version {
+			name = project.version
+			released = new Date()
+			//vscTag = '' //TODO
```
Comment (coderbot16) [inline build.gradle]: Should this be removed?

**Review by coderbot16 — CHANGES_REQUESTED** [review]
Overall looks good, you forgot some of your TODOs however!

File: .gitignore (hunk 1 above)
Comment (TheGlitch76) [reply .gitignore]: That was to protect against accidentally posting my bintray API key

File: build.gradle (hunk 2 above)
Comment (TheGlitch76) [reply build.gradle:1]: it's default indentation with fabric code style

Comment (coderbot16) [reply build.gradle:1]: maybe github is displaying them weird but they don't all look like tabs

File: .gitignore (hunk 1 above)
Comment (TheGlitch76) [reply .gitignore]: But it apparently didn't do anything lol

File: build.gradle [hunk 5]
```diff
@@ -62,10 +67,64 @@ tasks.withType(JavaCompile) {
...
+			}
+
+			// add all the jars that should be included when publishing to maven
+			afterEvaluate {
+				artifact jar
+				artifact sourcesJar
+			}
+		}
+	}
+
```
Comment (coderbot16) [inline build.gradle]: `blank lines`

**Review by coderbot16 — APPROVED** [review]
LGTM

**Pull request merged**